core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1478,6 +1497,23 @@ public void testDeleteIdentifierFieldColumnsFails() {
         () -> new SchemaUpdate(schemaWithIdentifierFields, SCHEMA_LAST_COLUMN_ID).deleteColumn("id").apply());
   }
 
+  @Test
+  public void testDeleteContainingNestedIdentifierFieldColumnsFails() {
+    Schema newSchema = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
+        .allowIncompatibleChanges()
+        .addRequiredColumn("out", Types.StructType.of(
+            Types.NestedField.required(SCHEMA_LAST_COLUMN_ID + 2, "nested", Types.StringType.get())
+        ))
+        .setIdentifierFields("out.nested")
+        .apply();
+
+    AssertHelpers.assertThrows("delete an identifier column without setting identifier fields should fail",
```

Comment:
nit: the message can be clearer like 'delete a struct with a nested identifier column without setting identifier fields should fail'?